server/plugin/api.go

```diff
@@ -314,11 +314,43 @@ func (p *Plugin) completeConnectUserToGitHub(c *Context, w http.ResponseWriter,
 		return
 	}
 
-	appErr = p.API.KVDelete(stateToken)
-	if appErr != nil {
-		p.API.LogWarn("Failed to delete state token", "error", appErr.Error())
-		http.Error(w, "error deleting stored state", http.StatusBadRequest)
-		return
+	if strings.Contains(stateToken, githubTokenKey) {
+		html := `
+		<!DOCTYPE html>
```

Comment:
This HTML content is duplicated in this function. Since we're not supposed to write a success response at this part of the function, I think we should just remove this string from this part of the function.

Also, I think it makes sense to have a function `createHTMLResponse(message string) string` that will return this HTML string with the `message` placed inside of the `<p>` tag.

Reply:
will pass an error as `invalid state` in this case